Backend/routes/listings.js

```diff
@@ -292,92 +302,101 @@ router.get("/search", async (req, res) => {
-    const notificationData = {
-      content: notificationContent,
-      userId: listing.sellerId,
-      isRead: false,
-      type: NotificationType.LIKE,
-      listingId: listing.id,
-      listingImage: listingImage,
-      usernameTarget: liker.username,
-    };
-    await sendNotification(notificationData, io);
+      const listingImage = listing.imageUrls[0];
 
-    res.status(201).json(like);
-  } catch (error) {
-    console.error("Error liking item:", error);
-    res
-      .status(500)
-      .json({ error: "Something went wrong while liking the item" });
+      const notificationContent = `@${liker.username} liked your item ${listing.title}.`;
+      const notificationData = {
+        content: notificationContent,
+        userId: listing.sellerId,
+        isRead: false,
+        type: NotificationType.LIKE,
+        listingId: listing.id,
+        listingImage: listingImage,
+        usernameTarget: liker.username,
+      };
+      await sendNotification(notificationData, io);
+
+      res.status(201).json(like);
+    } catch (error) {
+      console.error("Error liking item:", error);
+      res
+        .status(500)
+        .json({ error: "Something went wrong while liking the item" });
+    }
   }
-});
+);
 
-router.delete("/listings/:id/like", verifyToken, async (req, res) => {
-  const { id: itemId } = req.params;
-  const userId = req.user.id;
+router.delete(
+  "/listings/:id/like",
+  verifyToken,
+  async (req, res) => {
+    const { id: itemId } = req.params;
+    const userId = req.user.id;
 
-  try {
-    const existingLike = await prisma.like.findFirst({
-      where: {
-        userId: parseInt(userId),
-        itemId: parseInt(itemId),
-      },
-    });
+    try {
+      const existingLike = await prisma.like.findFirst({
+        where: {
+          userId: parseInt(userId),
+          itemId: parseInt(itemId),
+        },
+      });
 
-    if (!existingLike) {
-      return res.status(400).json({ message: "Item is not liked" });
-    }
+      if (!existingLike) {
+        return res.status(400).json({ message: "Item is not liked" });
+      }
 
-    await prisma.like.delete({
-      where: { id: existingLike.id },
-    });
+      await prisma.like.delete({
+        where: { id: existingLike.id },
+      });
 
-    res.status(200).json({ message: "Item unliked successfully" });
-  } catch (error) {
-    console.error("Error unliking item:", error);
-    res
-      .status(500)
-      .json({ error: "Something went wrong while unliking the item" });
+      res.status(200).json({ message: "Item unliked successfully" });
+    } catch (error) {
+      console.error("Error unliking item:", error);
+      res
+        .status(500)
+        .json({ error: "Something went wrong while unliking the item" });
+    }
   }
-});
+);
 
-router.get("/wishlist", verifyToken, async (req, res) => {
+router.get("/wishlist", verifyToken, logActivity(), async (req, res) => {
```

Comment:
are verifyToken and logActivity called sequentially? i would try to have the same order in every function

Reply:
Oh yes they are, just did some research and figured they are actually called sequentially and that is true.